### PR TITLE
MB-11322 document service item status

### PR DESCRIPTION
**Pull request by michaelvaldes-truss**
## Summary

This pr should add audit logging for approval and rejection of mto service items.
One thing to note is the refactor of the arguments passed to MoveHistoryDetailsSelector and PlainTextDetails; since the number of arguments has increased, I thought it might make more sense to make a shape for an audit history record and pass that instead of some of its fields.
Another thing to note is that the 'context' being built in the move history fetch query and passed in the move history payload is now a json object so that we can support more information in that field (in this case, both the shipment type and the service item name fetched from separate tables).

## Setup to Run Your Code

<details>
<summary>💻 You will need to use three separate terminals to test this locally.</summary>

##### Terminal 1

Start the Storybook locally.

```sh
make storybook
```

##### Terminal 2

Start the UI locally.

```sh
make client_run
```

##### Terminal 3

Start the Go server locally.

```sh
make server_run
```

</details>

### Additional steps

1. Access the office app as a TOO
2. Navigate to a move that is submitted or services counseling completed
3. Approve a service item and reject a service item
4. Verify that each service item update is shown in the audit history [as per the mapping](https://docs.google.com/spreadsheets/d/1r-c01NLG1j2KgdZAnmiECs5e5goySIoEGw2Ue36DGK4/edit#gid=569262977&range=17:19)

## Verification Steps for Author

These are to be checked by the author.

- [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
- [ ] Request review from a member of a different team.
- [x] Have the Jira acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

### Frontend

- [ ] User facing changes have been reviewed by design.
- [ ] There are no aXe warnings for UI.
- [ ] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/master/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
- [ ] There are no new console errors in the browser devtools
- [x] There are no new console errors in the test output
- [x] If this PR adds a new component to Storybook, it ensures the component is fully responsive, OR if it is intentionally not, a wrapping div using the `officeApp` class or custom `min-width` styling is used to hide any states the would not be visible to the user.

### Backend

- [ ] Code follows the guidelines for [Logging](https://transcom.github.io/mymove-docs/docs/dev/contributing/backend/Backend-Programming-Guide#logging)
- [ ] The requirements listed in [Querying the Database Safely](https://transcom.github.io/mymove-docs/docs/dev/contributing/backend/Backend-Programming-Guide/#querying-the-database-safely) have been satisfied.

#### Any new migrations/schema changes:

- [ ] Follows our guidelines for [Zero-Downtime Deploys](https://transcom.github.io/mymove-docs/docs/dev/contributing/database/Database-Migrations#zero-downtime-migrations)
- [ ] Have been communicated to #g-database
- [ ] Secure migrations have been tested following the instructions in our [docs](https://transcom.github.io/mymove-docs/docs/dev/contributing/database/Database-Migrations#secure-migrations)

### CODE REVIEW

File: pkg/services/move_history/move_history_fetcher.go [hunk 1]
```diff
@@ -38,27 +38,45 @@ func (f moveHistoryFetcher) FetchMoveHistory(appCtx appcontext.AppContext, param
...
 			AND audit_history.object_id = moves.id
 	),
+	service_items AS (
+		SELECT
+			mto_service_items.*, re_services.name, mto_shipments.shipment_type
+		FROM
+			mto_service_items
+		JOIN re_services ON mto_service_items.re_service_id = re_services.id
+		JOIN mto_shipments ON mto_service_items.mto_shipment_id = mto_shipment_id
+	),
```
Comment (donaldthai) [inline pkg/services/move_history/move_history_fetcher.go:65]: Could we add a `WHERE` clause here to only match with `move_id` from the `moves` subquery

File: src/constants/historyLogUIDisplayName.js [hunk 2]
```diff
@@ -72,21 +80,27 @@ export const shipmentOptionToDisplay = {
...
-    return changedValues.financial_review_flag === 'true'
-      ? 'Move flagged for financial review'
-      : 'Move unflagged for financial review';
-  },
-  requestShipmentCancellation: (_, oldValues) =>
-    `Requested cancellation for ${shipmentOptionToDisplay[oldValues.shipment_type]} shipment`,
-  updateMoveTaskOrderStatus: (changedValues) => {
-    return changedValues.status === 'APPROVED' ? 'Created Move Task Order (MTO)' : 'Rejected Move Task Order (MTO)';
-  },
+export const detailsPlainTextToDisplay = (historyRecord) => {
```
Comment (donaldthai) [inline src/constants/historyLogUIDisplayName.js:83]: I don't mind the changes to convert this into a function and switch statement. I'm sure @YanZ777 has some opinions here though :)

File: src/constants/historyLogUIDisplayName.js [hunk 3]
```diff
@@ -1,6 +1,14 @@
 import PropTypes from 'prop-types';
 
 export const HistoryLogValuesShape = PropTypes.object;
+export const HistoryLogContextShape = PropTypes.object;
+
+export const HistoryLogRecordShape = PropTypes.shape({
+  context: HistoryLogContextShape,
+  eventName: PropTypes.string,
+  changedValues: HistoryLogValuesShape,
```
Comment (YanZ777) [inline src/constants/historyLogUIDisplayName.js:9]: **suggestion (non blocking):** You could probably just get rid of the the `HistoryLogValuesShape` and the `HistoryLogContextShape` and just directly put `PropTypes.object` inside of this shape here.

File: src/constants/historyLogUIDisplayName.js (hunk 2 above)
Comment (michaelvaldes-truss) [reply src/constants/historyLogUIDisplayName.js:83]: @YanZ777 thanks for your input, I think the number of arguments for the components just grew unexpectedly XD

File: src/constants/historyLogUIDisplayName.js (hunk 3 above)
Comment (michaelvaldes-truss) [reply src/constants/historyLogUIDisplayName.js:9]: I thought so too, but our linting yells at me :(

File: pkg/services/move_history/move_history_fetcher.go (hunk 1 above)
Comment (michaelvaldes-truss) [reply pkg/services/move_history/move_history_fetcher.go]: Fixed!

File: src/constants/historyLogUIDisplayName.js (hunk 3 above)
Comment (YanZ777) [reply src/constants/historyLogUIDisplayName.js:11]: Aww dangit. 😢

**Review by YanZ777 — APPROVED** [review]
I'm good with the changes contained therein.

**Review by juliemoon — COMMENTED** [review]
The formatting matches the design specs. Another note: I don't know if this is a bug or if I'm not testing correctly but I see in the history log undefined and service items approved or rejected before I approved or rejected service items for the move. 
pre approved and rejected
undefined undefined

**Review by juliemoon — APPROVED** [review]
Synced with Michael and Donald. Comments will get addressed in another pr but from a design perspective it's good to go

**Pull request merged**